src/main/java/net/neoforged/moddevgradle/internal/PrepareRunOrTest.java

```diff
@@ -98,15 +98,16 @@ abstract class PrepareRunOrTest extends DefaultTask {
     /**
      * Only used when {@link #getRunTypeTemplatesSource()} is empty,
      * to know whether the associated Minecraft version requires one or two data runs.
+     * Defaults to latest.
```

Comment:
That's not correct. See `// Default to single run for backwards compatibility` that you removed.

Reply:
I'm pretty sure that neodev base will fail to find the `data` run type in 1.21.3 now. But I didn't test it.